zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Service.java

```diff
@@ -0,0 +1,85 @@
+package com.github.onsdigital.zebedee.api;
+
+import com.github.davidcarboni.cryptolite.Random;
+import com.github.davidcarboni.restolino.framework.Api;
+import com.github.onsdigital.zebedee.exceptions.NotFoundException;
+import com.github.onsdigital.zebedee.exceptions.UnauthorizedException;
+import com.github.onsdigital.zebedee.json.PermissionDefinition;
+import com.github.onsdigital.zebedee.model.ServiceAccount;
+import com.github.onsdigital.zebedee.permissions.service.PermissionsService;
+import com.github.onsdigital.zebedee.service.ServiceStore;
+import com.github.onsdigital.zebedee.service.ServiceStoreImpl;
+import com.github.onsdigital.zebedee.session.model.Session;
+import com.github.onsdigital.zebedee.session.service.SessionsService;
+import org.apache.commons.lang3.BooleanUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.POST;
+import java.io.IOException;
+import java.util.function.Supplier;
+
+import static com.github.onsdigital.zebedee.logging.ZebedeeLogBuilder.logInfo;
+
+@Api
+public class Service {
+
+    private Supplier<String> randomIdGenerator = () -> Random.id();
+
+    private ServiceStore serviceStore;
+
+    private SessionsService sessionsService;
+
+    private PermissionsService permissionsService;
+
+    @POST
+    public ServiceAccountWithToken createService(HttpServletRequest request, HttpServletResponse response) throws IOException, NotFoundException, UnauthorizedException {
+        final Session session = getSessionsService().get(request);
+        PermissionDefinition permissionDefinition = getPermissionsService().userPermissions(session.getEmail(), session);
+        if (BooleanUtils.isTrue(permissionDefinition.admin)) {
+            final ServiceStore serviceStoreImpl = getServiceStore();
+            final String token = randomIdGenerator.get();
+            ServiceAccount service = serviceStoreImpl.store(token, request.getInputStream());
```

Comment:
Do you need to put the input steam in a try-with-resources? Im never sure when its a request inputstream. Also can this throw an exception? If so we should handle the error case and log/return the appropriate response.

Reply:
The rule i always follow is, who ever creates the resources it should be there responsibly to clean up. As restolina creates the request, i would expect it to handle closing the stream.